apps/tests/lib/test_utils.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Tests.TestUtils do
+  alias ElixirChatbotCore.DocumentationManager.DocumentationFragment
+
+  @spec generate_output_name() :: String.t()
+  def generate_output_name do
+    DateTime.utc_now()
+    |> Timex.format!("{YYYY}-{0M}-{0D}-{h24}{m}{s}")
+  end
+
+  @spec fragments_to_html(DocumentationFragment.t()) :: String.t()
+  def fragments_to_html(fragments) do
+    fragments
+    |> Stream.map(fn fragment ->
+      """
+        <b> [#{fragment.type}] #{fragment.source_module}.#{fragment.function_signature} </b>
+        <div style="background-color: #f0f0f0"> #{Earmark.as_html!(fragment.fragment_text)} </div>
+      """
+    end)
+    |> concatenate_texts
+  end
+
+  def concatenate_texts(texts) do
+    texts
+    |> Enum.reduce("", fn string, acc -> acc <> string end)
```

Comment:
using `Enum.into` or `Enum.join` here should have better memory usage

Reply:
True, I didn't know about this function. Changed